kpi/utils/xml.py

```diff
@@ -154,3 +155,37 @@ def add_xml_declaration(xml_content: Union[str, bytes]) -> Union[str, bytes]:
     if use_bytes:
         return xml_.encode()
     return xml_
+
+
+def edit_submission_xml(
+    xml_parsed: etree._Element,
+    path: str,
+    value: str,
+) -> None:
+    """
+    Edit submission XML with an XPath and new value, creating a new tree
+    element if the path doesn't yet exist.
+    """
+    if '/' in path:
+        accumulated_elements = []
+        for i, element in enumerate(path.split('/')):
+            if i == 0:
+                if xml_parsed.find(element) is None:
+                    etree.SubElement(xml_parsed, element)
+                accumulated_elements.append(element)
+            else:
+                updated_xml_path = '/'.join(accumulated_elements)
+                if (
+                    xml_parsed.find(os.path.join(updated_xml_path, element))
+                    is None
+                ):
+                    etree.SubElement(xml_parsed.find(updated_xml_path), element)
+                accumulated_elements.append(element)
+
+    element_to_update = xml_parsed.find(path)
```

Comment:
You could rewrite the loop above so that the result of the last iteration's `find()` or new `SubElement` is stored; that'd avoid the need to find or create again after the loop.

Reply:
🚀